fractal/parser.py

```diff
@@ -373,9 +373,17 @@
     argument_default=ap.SUPPRESS,
     allow_abbrev=False,
 )
-task_edit_parser.add_argument(
-    "task_id_or_name", help="ID or name of task to edit", type=str
+
+task_edit_id_or_name_group = task_edit_parser.add_mutually_exclusive_group(
+    required=True
+)
+task_edit_id_or_name_group.add_argument(
+    "--id", help="ID of the task to edit", type=int
+)
+task_edit_id_or_name_group.add_argument(
+    "--name", help="Name of the task to edit", type=str
 )
+
```

Comment:
Add to version help message something like "(only accepted in combination with `--name`)"